aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -1,130 +1,139 @@
+        each_block_name = self._get_node_name(ast, each_block)
+        each_block_type = ast.get_type(each_block)
 
-                elif isinstance(each_block, IfStatement):
-                    self._check_if_statement(each_block, nested_level)
+        if each_block_type == ASTNodeType.METHOD_DECLARATION and each_block_name != self.method_name:
+            self._nested_methods(ast, each_block, nested_level)
 
-                elif type(each_block) in increment_for and type(each_block) in nested_for:
-                    self.complexity += 1 + nested_level
-                    self._traverse_childs(each_block, nested_level + 1)
+        elif each_block_type == ASTNodeType.IF_STATEMENT:
+            self._check_if_statement(ast, each_block, nested_level)
 
-                elif type(each_block) in increment_for and type(each_block) not in nested_for:
-                    if isinstance(each_block, BinaryOperation):
-                        if each_block.operator in logical_operators:
-                            self.complexity += 1
-                            self._increment_logical_operators(each_block, each_block.operator)
+        elif each_block_type in increment_for and each_block_type in nested_for:
+            self.complexity += 1 + nested_level
+            self._traverse_childs(ast, each_block, nested_level + 1)
 
-                    elif isinstance(each_block, MethodInvocation):
-                        is_recursion = self._is_recursion_call(each_block)
-                        self.complexity += is_recursion
+        elif each_block_type in increment_for and each_block_type not in nested_for:
+            if ast.get_type(each_block) == ASTNodeType.BINARY_OPERATION:
+                bin_operator = ast.get_binary_operation_name(each_block)
+                if bin_operator in logical_operators:
+                    self.complexity += 1
+                    self._increment_logical_operators(ast, each_block)
 
-                    else:
-                        self.complexity += 1
-                        self._traverse_childs(each_block, nested_level)
+            elif each_block_type == ASTNodeType.METHOD_INVOCATION:
+                is_recursion = self._is_recursion_call(ast, each_block)
+                self.complexity += is_recursion
 
-                else:
-                    self._traverse_childs(each_block, nested_level)
+            else:
+                self.complexity += 1
+                self._traverse_childs(ast, each_block, nested_level)
+        else:
+            self._traverse_childs(ast, each_block, nested_level)
+
+    def _get_node_name(self, ast, node) -> str:  # type: ignore
+        extracted_name = None
+        names = list(ast.children_with_type(node, ASTNodeType.STRING))
+        for each_string in names:
+            method_name = ast.get_attr(each_string, 'string')
+            if not method_name.startswith('/') and re.search(r'[^\W\d]', method_name) is not None:
```

Comment:
Can method name start from `/`? We better have a comment here to clarify what we are checking.

Reply:
added comments